client/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -82,7 +82,7 @@ trait Client[F[_]] {
   @deprecated("Use `Stream.eval(req).flatMap(client.stream).flatMap(f)`", "0.19.0-M4")
   def streaming[A](req: F[Request[F]])(f: Response[F] => Stream[F, A]): Stream[F, A]
 
-  def expectOr[A](req: Request[F])(onError: Response[F] => F[Throwable])(
+  def expectOr[A](req: Request[F])(onError: (Request[F], Response[F]) => F[Throwable])(
```

Comment:
This is a pretty aggressive compatibility change.

I believe you can pull the request Uri from the Response object, no?

Reply:
I don't see the reference to the orignal request inside the `Response`, Am I missing somehing?

Reply:
I don't think you can, but the caller has the request.  See below for an alternate idea.

Reply:
I've restored the old signature and used the approach suggested by @rossabaker